File: NN/Hole.py
from Vector2 import Vector2
from Ball import Ball

class Hole:

	pos = None
	radius = 0

	def __init__(self, x, y, r):
		self.pos = Vector2(x, y, 0)
		self.radius = r

	def score(self, ball):
		dist = self.pos.dist(ball.pos)
		return dist < self.radius

File: NN/js/hole.js
// @ts-check
"use strict";

class Hole {
    constructor(x, y, r) {
        this.pos = createVector(x, y);
        this.r = r;
    }

    draw() {
        stroke(8, 200, 27);
        fill(0);

        ellipse(this.pos.x, this.pos.y, this.r * 2);
    }

    score(ball)
    {
        let dist = this.pos.dist(ball.pos);
        return dist < this.r;
    }

    update() {

    }
}

if (typeof module !== 'undefined') {
    module.exports = Hole;
}
